cmd/thriftbreak/main_test.go

```diff
@@ -60,30 +88,67 @@ func TestThriftBreakIntegration(t *testing.T) {
 		"somefile.go": `service Qux{}`,
 	}
 	remove := []string{"test/d.thrift"}
-	breaktest.CreateRepoAndCommit(t, tmpDir, from, to, remove)
 
-	f, err := ioutil.TempFile(tmpDir, "stdout")
-	require.NoError(t, err, "create temporary file")
-	defer func(oldStdout *os.File) {
-		assert.NoError(t, f.Close())
-		os.Stdout = oldStdout
-	}(os.Stdout)
-	os.Stdout = f
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			tmpDir := t.TempDir()
+			breaktest.CreateRepoAndCommit(t, tmpDir, from, to, remove)
+
+			f, err := ioutil.TempFile(tmpDir, "stdout")
+			require.NoError(t, err, "create temporary file")
+			defer func(oldStdout *os.File) {
+				assert.NoError(t, f.Close())
+				os.Stdout = oldStdout
+			}(os.Stdout)
+			os.Stdout = f
 
-	err = run([]string{fmt.Sprintf("-C=%s", tmpDir)})
-	require.Error(t, err, "expected no errors")
-	assert.EqualError(t, err, "found 5 issues")
+			// Pinning not to run into scoping errors.
+			gitDir := tt.gitDirArg
+			extraCmd := tt.extraCmd
+			err = run([]string{fmt.Sprintf(gitDir, tmpDir), extraCmd})
 
-	stderr, err := ioutil.ReadFile(f.Name())
-	require.NoError(t, err)
+			require.Error(t, err, "expected no errors")
+			assert.EqualError(t, err, "found 5 issues")
 
-	out := string(stderr)
+			stderr, err := ioutil.ReadFile(f.Name())
+			require.NoError(t, err)
 
-	assert.Equal(t,
-		`c.thrift:deleting service "Baz"`+"\n"+
-			`d.thrift:deleting service "Qux"`+"\n"+
-			`v2.thrift:deleting service "Bar"`+"\n"+
-			`v1.thrift:removing method "methodA" in service "Foo"`+"\n"+
-			`v1.thrift:adding a required field "C" to "AddedRequiredField"`+"\n",
-		out)
+			out := string(stderr)
+			assert.Equal(t, tt.want, out)
+		})
+	}
+}
+
+func TestJsonPrinter(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		desc   string
+		want   string
+		writer func(io.Writer) func(compare.Diagnostic) error
+	}{
+		{
+			desc:   "json writer",
+			want:   "{\"File\":\"foo.thrift\",\"Message\":\"error\"}\n",
```

Comment:
Can use backticks here to avoid escaping most of the JSON.

```suggestion
			want:   `{"File":"foo.thrift","Message":"error"}` + "\n",
```